packages/kbn-expandable-flyout/src/hooks/use_sections_sizes.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+const RIGHT_SECTION_MIN_WIDTH = 380;
+const MIN_RESOLUTION_BREAKPOINT = 992;
+const RIGHT_SECTION_MAX_WIDTH = 750;
+const MAX_RESOLUTION_BREAKPOINT = 1920;
+
+const LEFT_SECTION_MAX_WIDTH = 1500;
+
+const FULL_WIDTH_BREAKPOINT = 1600;
+const FULL_WIDTH_PADDING = 48;
+
+export interface UserSectionsSizesParams {
+  /**
+   * The width of the browser window
+   */
+  windowWidth: number;
+  /**
+   * True if the right section is visible, false otherwise
+   */
+  showRight: boolean;
+  /**
+   * True if the left section is visible, false otherwise
+   */
+  showLeft: boolean;
+  /**
+   * True if the preview section is visible, false otherwise
+   */
+  showPreview: boolean;
+}
+
+export interface UserSectionsSizesResult {
+  /**
+   * Width of the right section in pixels
+   */
+  rightSectionWidth: number;
+  /**
+   * Width of the left section in pixels
+   */
+  leftSectionWidth: number;
+  /**
+   * Width of the flyout in pixels
+   */
+  flyoutWidth: string;
+  /**
+   * Left position of the preview section in pixels
+   */
+  previewSectionLeft: number;
+}
+
+/**
+ * Hook that calculate the different width for the sections of the flyout and the flyout itself
+ */
+export const useSectionSizes = ({
+  windowWidth,
+  showRight,
+  showLeft,
+  showPreview,
+}: UserSectionsSizesParams): UserSectionsSizesResult => {
+  // the right section's width will grow from 380px (at 992px resolution) to 750px (at 1920px resolution)
+  let rightSectionWidth: number = showRight
+    ? windowWidth < MIN_RESOLUTION_BREAKPOINT
+      ? RIGHT_SECTION_MIN_WIDTH
+      : RIGHT_SECTION_MIN_WIDTH +
+        (RIGHT_SECTION_MAX_WIDTH - RIGHT_SECTION_MIN_WIDTH) *
+          ((windowWidth - MIN_RESOLUTION_BREAKPOINT) /
+            (MAX_RESOLUTION_BREAKPOINT - MIN_RESOLUTION_BREAKPOINT))
+    : 0;
+
+  // handle tiny screens by not going smaller than the window width
+  if (rightSectionWidth > windowWidth) rightSectionWidth = windowWidth;
```

Comment:
I know small screen is not the main focus here, flagging it as observation: according to this line, if the windowWith is smaller than 380px, the right panel should take up the entire screen, during testing, I'm seeing a gap on the left which appears to be a `SecurityPageWrapper`

![image](https://github.com/elastic/kibana/assets/18648970/2979ce03-1bc9-4db5-9330-078886768d0c)

Reply:
good observation and I noticed this as well during testing. The code says it should take the whole width but the UI doesn't reflect that. I hadn't gone as far of trying to understand why, because I considered this a minor annoyance for a width that we don't fully support...

I think it's something we can keep in mind for future work, maybe. The bug existed before this refactor so I'm ok to not worry about this for now, if you are as well or course!